jgrapht-core/src/test/java/org/jgrapht/alg/clique/BaseBronKerboschCliqueFinderTest.java

```diff
@@ -0,0 +1,251 @@
+        set.add(V6);
+        set.add(V7);
+        expected.add(set);
+
+        set = new HashSet<>();
+        set.add(V3);
+        set.add(V4);
+        set.add(V5);
+        expected.add(set);
+
+        set = new HashSet<>();
+        set.add(V7);
+        set.add(V8);
+        expected.add(set);
+
+        set = new HashSet<>();
+        set.add(V1);
+        set.add(V2);
+        set.add(V9);
+        set.add(V10);
+        expected.add(set);
+
+        // convert result from Collection to Set because we don't want
+        // order to be significant
+        Set<Set<String>> actual = new HashSet<>(cliques);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNonSimple()
+    {
+        Graph<String, DefaultEdge> g = new Pseudograph<>(DefaultEdge.class);
+        g.addVertex("1");
+        g.addVertex("2");
+        g.addEdge("1", "2");
+        g.addEdge("1", "2");
+
+        MaximalCliqueEnumerationAlgorithm<String, DefaultEdge> finder = createFinder1(g);
+        Iterator<Set<String>> it = finder.iterator();
+        while (it.hasNext()) {
+            it.next();
+        }
+    }
+
+    public static void createGraph(Graph<String, DefaultEdge> g)
+    {
+        g.addVertex(V1);
+        g.addVertex(V2);
+        g.addVertex(V3);
+        g.addVertex(V4);
+        g.addVertex(V5);
+        g.addVertex(V6);
+        g.addVertex(V7);
+        g.addVertex(V8);
+        g.addVertex(V9);
+        g.addVertex(V10);
+
+        // biggest clique: { V1, V2, V3, V4 }
+        g.addEdge(V1, V2);
+        g.addEdge(V1, V3);
+        g.addEdge(V1, V4);
+        g.addEdge(V2, V3);
+        g.addEdge(V2, V4);
+        g.addEdge(V3, V4);
+
+        // smaller clique: { V5, V6, V7 }
+        g.addEdge(V5, V6);
+        g.addEdge(V5, V7);
+        g.addEdge(V6, V7);
+
+        // for fun, add an overlapping clique { V3, V4, V5 }
+        g.addEdge(V3, V5);
+        g.addEdge(V4, V5);
+
+        // make V8 less lonely
+        g.addEdge(V7, V8);
+
+        // add one more maximal which is also the biggest { V1, V2, V9, V10 }
+        g.addEdge(V1, V9);
+        g.addEdge(V1, V10);
+        g.addEdge(V2, V9);
+        g.addEdge(V2, V10);
+        g.addEdge(V9, V10);
+    }
+
+    @Test
+    public void testTimeout()
+    {
+        final int size = 100;
+        Graph<Object, DefaultEdge> g = new SimpleGraph<>(DefaultEdge.class);
+        CompleteGraphGenerator<Object, DefaultEdge> completeGraphGenerator =
+            new CompleteGraphGenerator<>(size);
+        completeGraphGenerator.generateGraph(g, new ClassBasedVertexFactory<>(Object.class), null);
+
+        MaximalCliqueEnumerationAlgorithm<Object, DefaultEdge> finder =
+            createFinder2(g, 1, TimeUnit.SECONDS);
+
+        Set<Set<Object>> cliques = new HashSet<>();
+        finder.iterator().forEachRemaining(cliques::add);
+
+        int found = cliques.size();
+        assertTrue(found == 0 || found == 1);
```

Comment:
I'm not sure how valuable this test is? This test works on the assumption that the machine which executes the clique finder finds 0 or 1 cliques within the time limit. The faster our test machines become, the more often it will find 1. I would just remove this test. Or, if you really want to test this, put a time limit on the computations, and use an external timer to measure how long the computations take (and whether the time limit was respected). 
Personally I wouldn't bother. If this breaks, then I'm sure we'll get complaints. This issue would be easy to troubleshoot.